### ISSUE TITLE
Aleksandra: State, week 1

### ISSUE THREAD

**Issue by aleks2407** [HackYourFutureBelgium/class-9-10#603]
## Suggested Study
- [x] What is state and why should we care about it?
- [x] /integrate/stepped-and-separated (code to study)

## Exercises
- [ ] rendering-data

---

## Week 1

- [x] Sunday Prep

- [x] Sunday Class

- [ ] [Group Project](https://github.com/HackYourFutureBelgium/class-9-10/issues/602)

### Progress

I was studying the code from the integrate exercises as well the DOM exercises. While you read the ready code, you have a feeling that you kind of understand it. But when it comes to recreating this code on your own, for now it seems to be very difficult for me.

### Blocked

This week's project looks complex and difficult for our group so far. We made two Zoom calls in terms of splitting the work and 
planning the project. But it didn't seem to help much to be honest. I have a luxury of having a friend who is a programmer. He had time and he helped me to explain his view of the project and helped with the code. But without this help, I would probably spend at least two weeks on coming to something similar. 
The split of work gets more complicated because on this level , we are not able to see what exactly should be in the code without having a tutorial with a ready code to follow. 

### Next Steps

To bring the project to life)) if possible. To work further on the exercises and other material, there is so much of it that you never honestly can say that you are DONE)))

**Comment by colevandersWands** [comment]
> While you read the ready code, you have a feeling that you kind of understand it. But when it comes to recreating this code on your own, for now it seems to be very difficult for me.

You're not the only one with this concern, it's totally normal.  @oguzkarademir said the same thing [in his check-in](https://github.com/HackYourFutureBelgium/class-9-10/issues/601).  Something you can try is to copy-paste from an exercise you did understand and then to modify it for the project you're working on

**Comment by colevandersWands** [comment]
> This week's project looks complex and difficult for our group so far. We made two Zoom calls in terms of splitting the work and planning the project. But it didn't seem to help much to be honest.

@aleks2407 I see your team is having good discussions in your channel, this project is harder than I'd intended.  Would it help if you had two weeks to finish it?

**Comment by aleks2407** [comment]
> > This week's project looks complex and difficult for our group so far. We made two Zoom calls in terms of splitting the work and planning the project. But it didn't seem to help much to be honest.
> 
> @aleks2407 I see your team is having good discussions in your channel, this project is harder than I'd intended. Would it help if you had two weeks to finish it?

I'm not sure))) I think for me personally it would definitely help, if I write all the code myself. But splitting it into parts that will be written independently by 5 people. That's the struggle.

**Comment by colevandersWands** [comment]
What if gave you the parts that need to be written as empty handler/view/logic functions?  that way you could all focus more on the coding than on the dividing?

I'm trying to figure out the best balance of guidance and letting you figure it out on your own.  Do you have any ideas or suggestions?

**Comment by aleks2407** [comment]
@colevandersWands  I got help from my friend this week with the project. So I did the split. But others don't understand it and don't know how to go further. If you could look on the split I posted on the Slack in Group 4? I don't know how to approach this differently. When we just followed a tutorial two previous weeks, we had the code ready, so everyone knew what to write. This week is very different, all the code is connected, so it's not like creating 6 apart buttons that have their own functionality and just use the same data to process. 
It just looks a very big step in comparing what we did before.

**Comment by colevandersWands** [comment]
I've seen the split in your channel, it looked good. But as you said, it's not something you could have come up with on your own which means the project was indeed too big of a step.

What would be a better way for you to have taken this step?  it's not too late to adjust this week and next week's projects.  I see two ways to go:
1. continue giving you tutorials to reverse-engineer
2. giving you a development strategy and minimal starter code, but no tutorial

From what you're saying I think 2 is the best option?   this would give you some more practice writing code from scratch without needing to lose time planning the project, which was the focus of last module's projects anyway.

**Comment by aleks2407** [comment]
@colevandersWands yes, absolutely!The second option seems to be giving more chance to research (while researching you learn more than while seeing a code done by someone else ready)

**Comment by colevandersWands** [comment]
I'll write up a dev-strat and fill in some starter code for the quiz tomorrow after getting a better idea where you're all at in the review call.   

For next week I'll create a simpler project so you have a chance to catch up on this one

**Comment by aleks2407** [comment]
Thank you @colevandersWands, this will be super helpful!! I shared as well with group a link to this issue, so they can follow too.
